dali/ft/filecopy.cpp

```diff
@@ -2637,15 +2640,11 @@ void FileSprayer::setSourceTarget(IFileDescriptor * fd, DaftReplicateMode mode)
         setCopyCompressedRaw();
 }
 
-void FileSprayer::setTargetCompression(bool compress)
-{
-    compressOutput = compress;
-}
-
 void FileSprayer::setTarget(IDistributedFile * target)
 {
     distributedTarget.set(target);
-    compressOutput = !encryptKey.isEmpty()||target->isCompressed();
+    LOG(MCdebugInfo, unknownJob, "FileSprayer::setTarget: compressedInput:%d, compressOutput:%d, preserveCompression:%d ", compressedInput, compressOutput, preserveCompression);
+    compressOutput = !encryptKey.isEmpty() || target->isCompressed() || (preserveCompression && compressedInput);
```

Comment:
I'm not sure this is right.
I would imply that the IDistributeFile's meta data (that's being published)compressed, is not compressed (i.e. target->isCompressed() == false).

In tests was the published compression meta info correct?

If it were correct coming into setTarget, then (preserveCompression && compressedInput) wouldn't be needed.

I think should bet set before, as the target IDistributeFile is created.

Reply:
Yes, it seems to use target->isCompressed() was not correct. 
However, there are two things:
1. The compress parameter doesn't populated through the chain therefore it is always false
2. Instead of 1. the compression is applied to the target file in CFileSprayEx::onCopy() and CFileSprayEx::onSprayXXXX (in ws_fsservice.cpp).
So at this point the target file compression is already set (that's why I used it, but it is a bit misleading). 
I can see two solution:
1. Restore the compress option from the target file (with isCompressed()) actually this is implemented in this push.
2. Implement the compress parameter passing through the request/call chain.